src/python/pants/notes/2.15.x.md

```diff
@@ -1,5 +1,66 @@
 # 2.15.x Release Series
 
+## What's New
+
+Pants 2.15 brings support for several new features:
+
+### Environments
+
+Pants now supports alternate execution environments for build actions. These "environments" can be explicitly
+configured as desired. Pants previously supported local execution (which is now called a `local_environment`)
+and remote execution via the Remote Execution API (which is now called a `remote_environment`). Pants 2.15
+also supports executing build actions within a Docker container via the `docker_environment` target type.
+(Note: `docker_environment` is a distinct concept from the Docker backend which actually builds images.)
+
+### New goal: `fix`
+
+The `fix` goal has been added to run tools which generally will perform more invasive changes on your source
+files than the `fmt` goal. The `fmt` goal remains for tools which merely format code without making
+semantic changes.
+
+### Backends
+
+#### New: OpenAPI
+
+The new OpenAPI backend provides support for linting OpenAPI API specifications using the `spectral` tool
+and generating Java code from the OpenAPI specifications.
+
+Enable the `pants.backend.experimental.openapi` backend for base OpenAPI support,
+`pants.backend.experimental.openapi.lint.spectral` for linting support, and
+`pants.backend.experimental.openapi.codegen.java` for Java code generation support.
+
+#### Go 
+
+The Go backend received several minor changes including:
+
+- Support for the `golangci-lint` linter.
+- Support for multiple `go_mod` targets in same repository.
+- A new `go-generate` custom goal to run `go generate` on any `//go:generate` instructions in a package.
+- `$GOROOT/bin` is now prepended to `PATH` when running tests to enable them to easily find the `go` binary for
+the Go toolchain in use.
+
+Note: Due to certain changes in Go v1.20, Pants 2.15.x and earlier Pants versions will not work with Go v1.20
+and later releases. Instead, use Pants v2.16.x or later Pants versions to support Go v1.20 and later releases.
+
+### Plugin API changes
+
+Several changes and additions were made to the Plugin API:
+
+#### Formatters
+
+The schema used for formatters and linters has been changed in a backwards incompatible way in order to allow for
+formatters that do not require a target to operate.
+
+#### Synthetic Targets
+
+Plugins may now add "synthetic targets" to the build graph to create freestanding targets programmatically. 
+These synthetic targets can also be generator targets in order to further create generated targets.
+
+#### Batched Tests
+
+The `test` goal API has been updated to support explicit batching of tests.
```

Comment:
I'm wondering if this should be higher up the list, maybe right after environments, as it is probably the most impactful on the user. Also we probably want to explain why this is cool, namely that it can improve performance in cases where test setup is too expensive to run separately on each test file.

Reply:
It should be in both places. The Plugin API changes should be here, while the user-facing aspects of test batching can be earlier. I don't have context on the user-facing aspects. Which backend(s) actually make use of test batching? Just the Python backend?

Reply:
I've added a separate section for the user-facing part of test batching. Plus added detail to other sections including links into the 2.15 documentation.